Architecture.md

```diff
@@ -71,6 +71,79 @@
+
+##### Creating and deleting repositories
+
+When a user initializes a new project, a new repository has to be created, it also has to be deleted when the project is deleted.
+
+The repository is created using the [git init](https://git-scm.com/docs/git-init) command. The repository is then stored in a directory, which is named after the **project's id**. It is stored on the host machine.
+
+For now, it is not scalable, as the git repositories are stored on the host machine. In future version, the storage could be moved to a dedicated storage server, a distrubuted storage system or a bucket storage system.
+
+In that sense, to prepare for future versions, the git controller is a separated component from the rest of the application, designed to be stateless, and only stores the repositories on the host machine. It does not store any information about the repositories, and only uses the project's id to name the repository's directory. This way, the git controller can be scaled horizontally, and, if needed, the repositories can be stored on a dedicated storage server after adding a layer of authorization.
+
+Since it is separated it needs to communicate with the rest of the application. It was achieved using [gRPC](https://grpc.io/), which is a high performance RPC framework. It is used to create a bidirectional stream between the git controller and the rest of the application. The git controller can then receive commands from the rest of the application, and send back the result of the command.
+
+We made the choice to use gRPC instead of a REST API. Indeed, since it uses [protobuf](https://developers.google.com/protocol-buffers), it is easier to maintain and to evolve as the application grows. It is also more performant than a REST API, and is easier to use, since it is strongly typed. It's support on different languages also makes it easier to use. It also limits communication problematics between teams and services, since the communication is well defined and documented through the protobuf file.
+
+It was made in rust for its high performance, reliability, robustness and memory safety using the [tonic](https://github.com/hyperium/tonic) framework, which is a rust implementation of gRPC. It was chosen because it is the most used gRPC framework in rust, and is well documented.
+
+##### Authentication
+
+Once the repositories are stored, the clients need to be able to push to them.
+
+In order to let the users push to thoses repositories, we need to give them access to the host machine. An easy solution would be to give them a ssh access to the host machine with servers like [sshd](https://www.openssh.com/). However, this would give them access to the whole host machine, which is not what we want. Indeed, we want to limit the access of the clients to their own repositories, and not to the whole host machine. And pose a great security risk.
```

Comment:
`And pose a great security risk.` What does this mean? Has this been left out?

Reply:
yep

fix: 8e2d312eab5b72e85448f74f8bd8976518f9ac46